core/libcamera_app.hpp

```diff
@@ -119,6 +119,12 @@ class LibcameraApp
 	Stream *LoresStream(StreamInfo *info = nullptr) const;
 	Stream *GetMainStream() const;
 
+	CameraManager *GetCameraManager() const;
+	std::vector<std::shared_ptr<libcamera::Camera>> GetCameras()
+	{
+		return GetCameras(camera_manager_);
+	}
+
```

Comment:
I don't think the ``GetCameras()`` override is needed right?  Instead you need to change (below):
```
static std::vector<std::shared_ptr<libcamera::Camera>> GetCameras(const std::unique_ptr<CameraManager> &cm)
```
to
```
static std::vector<std::shared_ptr<libcamera::Camera>> GetCameras(const CameraManager> *cm)
```

Reply:
right. 

But I already have a LibCameraApp, so I might as well just use a new app->GetCameras() off of it and not deal with the camera manager at all? After all, I was only doing that because that's what the multiple other code pieces in LibCameraApp did and you didn't have a singular mens to get at the list.

Also, you'd need to change "public", if interim, API.. I don't care one way or another, but given there's no way to get at the manager without an app in the first place, and there's not way to get a unique_ptr without having one first, I would recommend (again without knowing a lot about what all of this does or should do) to remove the static method and just have an instance method....

Reply:
OK, so the options use that method... why? I guess this is where our problems came from, as we're using the options when the app has already been created. So now you have two cam managers... at least I think so.

Reply:
There is another place that uses GetCameras() with the camera manager already opened.  This is why passing in an external camera manager must still be supported.